widgets/js/index.tsx

```diff
@@ -210,3 +214,8 @@ function renderHistogram({ model, el }: RenderArgs) {
   model.on("change:buckets", onChange);
   model.on("change:shot_count", onChange);
 }
+
+function renderCircuit({ model, el }: RenderArgs) {
+  const circuitJson = model.get("circuit_json") as string;
+  prender(<Circuit circuit={JSON.parse(circuitJson)}></Circuit>, el);
```

Comment:
If you follow the above pattern with rendering in `onChange` and an event handler, then you get updates to the rendering for free (i.e. if the user holds a reference to the widget and sets a new circuit trace on it, it will update the circuit drawing in place)

Reply:
Makes sense, I'll add the handler. It wasn't clear to me why we'd _want_ that kind of dynamic behavior, but I think I just don't have much experience with widgets.